### PR TITLE
test: add failing bfloat16 test case for metal backend

### PR DESCRIPTION
**Pull request by skoshx**
Created a failing test for Metal for issue #3453

### CODE REVIEW

File: test/test_ops.py [hunk 1]
```diff
@@ -1624,6 +1624,11 @@ def test_one_hot(self):
     data = [[[1, 2, 3], [0, 3, 5]], [[1, 2, 3], [0, 3, 5]]]
     helper_test_op([], lambda: torch.nn.functional.one_hot(torch.tensor(data), 8), lambda: Tensor(data).one_hot(8), forward_only=True)
 
+  @unittest.skipIf(getenv("METAL", 0)== 0, "only run on METAL backend")
```
Comment (chenyuxyz) [inline test/test_ops.py:1627]: I think bfloat16 tests in test_dtype.py is a better place to add this. What's special about uniform? Would be nice to have the minimum test case that shows the issue. Finally need to set expect failure instead of just letting it fail.

Comment (skoshx) [reply test/test_ops.py:1627]: Hey, I can make it simpler with maybe just like `Tensor([1])`, nothing special about uniform.

And isn't letting it fail correct? I'm assuming that these errors aren't the expected output? Shouldn't tinygrad somehow gracefully handle `bfloat16` on Metal backend?

If I make it expect failure, then the CI essentially reflects that when there are bugs, the CI should pass?

**Comment by skoshx** [comment]
@chenyuxyz moved to `test_dtype.py`. Funnily enough, just using something like `Tensor([1,2,3])` or `Tensor.eye(3)` doens't reproduce the bug, so the uniform seems to be the simplest to reproduce it.

Also added `@unittest.expectedFailure`

**Comment by chenyuxyz** [comment]
running
```
from tinygrad import Tensor
x = Tensor.uniform(10, dtype=dtypes.bfloat16)
x.realize()
```
has no problem on my local M1 Max. Is the test set up correctly?

**Comment by skoshx** [comment]
Weird, fails for me on local M2 Pro. And yeah I think so, earlier when the test suite ran without the `expectedFailure`, my test correctly failed on the CI.

**Comment by skoshx** [comment]
Screenshot 2024-03-05 at 17 25 16

I'm suspecting this is just some super weird bug on my machine. Since M1, M2, M3 should all support `bfloat16` according to [Metal docs](https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf).
